Lib/fontTools/varLib/__init__.py

```diff
@@ -713,20 +733,27 @@ def load_designspace(designspace):
 		])
 
 	# Setup axes
+	if not ds.axes:
+		raise VarLibValidationError(f"Designspace must have at least one axis.")
+
 	axes = OrderedDict()
-	for axis in ds.axes:
+	for axis_index, axis in enumerate(ds.axes):
 		axis_name = axis.name
 		if not axis_name:
-			assert axis.tag is not None
+			if not axis.tag:
+				raise VarLibValidationError(f"Axis at index {axis_index} needs a tag.")
 			axis_name = axis.name = axis.tag
+			# TODO: Set axis_name to the corresponding default name from
+			# standard_axis_map so the code below gives it a default name.
```

Comment:
i don't understand this TODO

Reply:
If `axis.name` is unset and we're setting it to the axis tag string, we miss fetching the label name in the `if axis_name in standard_axis_map` conditional.

The follwoing code does it:
```python
	# Provide fallback standard name table entries for known axes.
	standard_names = {
		"weight": ("wght", {"en": "Weight"}),
		"width": ("wdth", {"en": "Width"}),
		"slant": ("slnt", {"en": "Slant"}),
		"optical": ("opsz", {"en": "Optical Size"}),
		"italic": ("ital", {"en": "Italic"}),
	}
	tag_to_standard_names = {v[0]: k for k, v in standard_names.items()}

	# Set up axes.
	# [...]

	axes = OrderedDict()
	for axis_index, axis in enumerate(ds.axes):
		if not axis.name:
			if not axis.tag:
				raise VarLibValidationError(f"Designspace {ds_path}: Axis at index {axis_index} needs a tag.")
			axis.name = tag_to_standard_names[axis.tag]  # Fill in default name below.

		if axis.name in standard_names:
			if axis.tag is None:
				axis.tag = standard_names[axis.name][0]
			if not axis.labelNames:
				axis.labelNames.update(standard_names[axis.name][1])
		# [...]
	# [...]
```

Maybe this can be refactored even more.

Reply:
the whole `standard_axis_map` is kind of legacy, predates the time DS axes were given an explicit `tag` and `labelNames` attributes. Keep as is, and remove the TODO comment.